test/t8_cmesh/t8_gtest_cmesh_face_is_boundary.cxx

```diff
@@ -0,0 +1,188 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <gtest/gtest.h>
+#include <t8_cmesh.h>
+#include <t8_cmesh/t8_cmesh_examples.h>
+
+/* Test for one tree*/
+/* *INDENT-OFF* */
+class cmesh_face_boundary_one_tree : public testing::TestWithParam<t8_eclass>{
+protected:
+  void SetUp() override {
+    eclass = GetParam();
+
+    /* For each eclass create a cmesh consisting only of one tree.
+     * We then check whether all faces of this tree are a boundary face. */
+    cmesh = t8_cmesh_new_from_class (eclass, sc_MPI_COMM_WORLD);
+  }
+  void TearDown() override {
+    t8_cmesh_destroy(&cmesh);
+  }
+
+  t8_cmesh_t          cmesh;
+  t8_eclass           eclass;
+  int                 num_faces;
+  int                 iface;
```

Comment:
This only used as an iterator in the for-loops. You can declare and initialize the variable in the for-loop via:
`for(int iface=1.....)`. That way, you can only use iface in the for-loop and not use it by accident somewhere else, which is a little safer.